### PR TITLE
[Wallet] Send Amount Loading Spinner pt 2

### PR DESCRIPTION
**Pull request by tarikbellamine**
### Description
Fixed a bug in PR #4924 and added a test to ensure functionality is correct.

### CODE REVIEW

File: packages/mobile/src/send/SendAmount.tsx [hunk 1]
```diff
@@ -91,11 +90,28 @@ export const sendAmountScreenNavOptions = ({
...
+  const localCurrencyExchangeRate = useSelector(getLocalCurrencyExchangeRate)
+  const localCurrencySymbol = useSelector(getLocalCurrencySymbol)
+  const e164NumberToAddress = useSelector(e164NumberToAddressSelector)
+  const dollarBalance = useSelector(stableTokenBalanceSelector)
+  const recipientVerificationStatus = getRecipientVerificationStatus(recipient, e164NumberToAddress)
+  const feeType = getFeeType(recipientVerificationStatus)
+  const estimateFeeDollars = useSelector(getFeeEstimateDollars(feeType))
+
+  const dispatch = useDispatch()
+
```
Comment (tarikbellamine) [inline packages/mobile/src/send/SendAmount.tsx:110]: All the above is the same except for `reviewButtonInnerElement`

File: packages/mobile/src/send/SendAmount.tsx [hunk 2]
```diff
@@ -108,20 +124,20 @@ function SendAmount(props: Props) {
     dispatch(fetchAddressesAndValidate(recipient.e164PhoneNumber))
   }, [])
 
-  const { t } = useTranslation(Namespaces.sendFlow7)
+  useEffect(() => {
```
Comment (tarikbellamine) [inline packages/mobile/src/send/SendAmount.tsx:124]: Most the new stuff

File: packages/mobile/src/send/SendAmount.tsx [hunk 3]
```diff
@@ -173,14 +189,7 @@ function SendAmount(props: Props) {
...
   const isAmountValid = parsedLocalAmount.isGreaterThanOrEqualTo(DOLLAR_TRANSACTION_MIN_AMOUNT)
-  const isDollarBalanceSufficient = isAmountValid && newAccountBalance.isGreaterThanOrEqualTo(0)
-
-  const reviewButtonInnerElement =
-    reviewButtonPressed && fetchingValidationStatus ? (
-      <ActivityIndicator testID={'loading/sendAmount'} />
-    ) : (
-      t('global:review')
-    )
+  const isDollarBalanceSufficient = isAmountValid && newAccountBalance.isGreaterThan(0)
```
Comment (tarikbellamine) [inline packages/mobile/src/send/SendAmount.tsx:184]: Needs to be greater than or the user definitely won't have enough to pay for gas

File: packages/mobile/src/send/SendAmount.tsx [hunk 4]
```diff
@@ -332,7 +325,7 @@ function SendAmount(props: Props) {
         text={reviewButtonInnerElement}
         type={BtnTypes.SECONDARY}
         onPress={onReviewButtonPressed}
-        disabled={!isAmountValid}
+        disabled={!isAmountValid || reviewButtonPressed}
```
Comment (tarikbellamine) [inline packages/mobile/src/send/SendAmount.tsx:327]: Added this so the button can't be pressed while the loading spinner is present

File: packages/mobile/src/send/SendAmount.tsx [hunk 5]
```diff
@@ -91,11 +90,28 @@ export const sendAmountScreenNavOptions = ({
...
 
 function SendAmount(props: Props) {
-  const dispatch = useDispatch()
+  const { t } = useTranslation(Namespaces.sendFlow7)
+
+  const [amount, setAmount] = useState('')
+  const [reviewButtonPressed, setReviewButtonPressed] = useState(false)
+  const [reviewButtonInnerElement, setReviewButtonInnerElement] = useState<string | JSX.Element>(
+    t('global:review')
+  )
```
Comment (jeanregisser) [inline packages/mobile/src/send/SendAmount.tsx]: nit: from what I see, this state could be removed and we could derive it from `reviewButtonPressed` and `recipientVerificationStatus`.

**Review by jeanregisser — APPROVED** [review]
Thanks for fixing this! 👍 
Left one optional comment.

File: packages/mobile/src/send/SendAmount.tsx (hunk 5 above)
Comment (tarikbellamine) [reply packages/mobile/src/send/SendAmount.tsx]: Good point. That will save us a rerender.

**Comment by Lss-Ankit** [comment]
Hi @jeanregisser i have verified task using Test flight build v1.1.0 (20) and Android play store build v1.1.0 (1004294308) and found below new issue:

**Test scenarios:**

1. Verify user is able to see the send button on home page 
2. Verify user redirect to Send contacts page after tapping on send button
3. Verify user should able to search any contact 
4. Verify user is able to select any searched contact
5. Verify user should redirect to send page after selecting any contact
6. Verify user is able to see the review button on Send page 
7. Verify Review button should show Disable when user enter invalid amount 
8. verify Review button should show enable when user enter valid amount 
9. Verify user is able to see the spinner loading indicator when tapping on Review button (Issue: [#5019](https://github.com/celo-org/celo-monorepo/issues/5019) )
10. Verify user is already selected then loading should not shown and review button enable immediately after entering amount
11. Verify user should redirect to sending page after tapping on review button 

**New issues:**

-  [#5019](https://github.com/celo-org/celo-monorepo/issues/5019)

**Note:** we are blocked to perform this task on android because app get crash when user tap on Review button 

Please let me know if any thing else i need to verify for this task.
Thanks !

**Pull request merged**